File: Easy/2180_Count_Integers_With_Even_Digit_Sum/Solution.py


class Solution:
    def sumOfDrgit(self, n):
        ans = 0
        while n > 0:
            ans += n % 10
            n //= 10
        return ans
    
    def countEven(self, num: int) -> int:
        ans = 0

        for i in range(1, num + 1):
            if self.sumOfDrgit(i) % 2 == 0:
                ans += 1

        return ans


if __name__ == "__main__":
    solution = Solution()
    
    # test cases 1
    print(solution.countEven(4))

    # test cases 2
    print(solution.countEven(30))
        
        

File: Easy/2180_Count_Integers_With_Even_Digit_Sum/Solution.java


public class Solution {
    private int sumOfDrgit(int n) {
        int sum = 0;
        while (n > 0) {
            sum += n % 10;
            n /= 10;
        }
        return sum;
    }

    public int countEven(int num) {
        int ans = 0;

        for (int i = 2; i <= num; i++) {
            int n = sumOfDrgit(i);
            if (n % 2 == 0) ans++;
        }

        return ans;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        System.out.println(solution.countEven(4));

        // test cases 2
        System.out.println(solution.countEven(30));
    }
}